kernel/kernel-defaults/src/test/java/io/delta/kernel/defaults/benchmarks/workloadrunners/WriteRunner.java

```
@@ -0,0 +1,179 @@
+          currentSnapshot.buildUpdateTableTransaction("Delta-Kernel-Benchmarks", Operation.WRITE);
+
+      Transaction txn = txnBuilder.build(engine);
+      Row txnState = txn.getTransactionState(engine);
+      DataWriteContext writeContext =
+          Transaction.getWriteContext(engine, txnState, new HashMap<>() /* partitionValues */);
+      CloseableIterator<Row> add_actions =
+          Transaction.generateAppendActions(
+              engine, txnState, toCloseableIterator(actions.iterator()), writeContext);
+      TransactionCommitResult result =
+          txn.commit(engine, CloseableIterable.inMemoryIterable(add_actions));
+
+      long version = result.getVersion();
+      blackhole.consume(version);
+
+      // Use the post-commit snapshot for the next transaction
+      // Post-commit snapshot should always be present unless there was a conflict
+      currentSnapshot =
+          result
+              .getPostCommitSnapshot()
+              .orElseThrow(
+                  () ->
+                      new IllegalStateException(
+                          "Post-commit snapshot not available. This indicates a conflict "
+                              + "occurred during the benchmark, which should not happen. "
+                              + "Ensure no other processes are writing to the table: "
+                              + workloadSpec.getTableInfo().getResolvedTableRoot()));
+    }
+  }
+
+  /** Cleans up the state created during benchmark execution by reverting all committed changes. */
+  @Override
+  public void cleanup() throws Exception {
+    if (initialDeltaLogFiles == null) {
+      return; // Setup didn't complete, nothing to clean up
+    }
+    // Delete any files that weren't present initially
+    Set<String> currentFiles = captureFileListing();
+    for (String filePath : currentFiles) {
+      if (!initialDeltaLogFiles.contains(filePath)) {
+        engine.getFileSystemClient().delete(filePath);
```

Maybe swallow NOT_FOUND error here?